src/main/java/com/google/starfish/servlets/GetTrendingNotesServlet.java

```diff
@@ -0,0 +1,63 @@
+package com.google.starfish.servlets;  
+
+import java.io.IOException;  
+import java.sql.SQLException;  
+import javax.servlet.ServletException;  
+import javax.servlet.annotation.WebServlet;  
+import javax.servlet.http.HttpServlet;  
+import javax.servlet.http.HttpServletRequest;  
+import javax.servlet.http.HttpServletResponse;  
+import javax.sql.DataSource;
+import com.google.starfish.services.FavoriteNoteService;
+import com.google.starfish.models.Note;
+import com.google.gson.Gson;
+
+/** Servlet that returns trending notes based on a given timespan as a query param */
+@WebServlet("/get-trending-notes")  
+public class GetTrendingNotesServlet extends HttpServlet {  
+
+  private FavoriteNoteService favoriteNoteService = new FavoriteNoteService();
+
+  @Override
+  public void doGet(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");  
+
+    String timespan = trimAndLowerCaseString(req.getParameter("timespan"));
+    // Default timespan is all-time
+    if (timespan == null) timespan = "all-time";
+    try {
+      Object[][] trendingNotes = null;
+      switch(timespan) {
+        case "today":
```

Comment:
looks like you re-use these below so would be safer to use an enum to keep consistent

Reply:
+1

Reply:
Discussed offline -- we can use the existing Recency enum in the FavoriteNoteService class to enforce consistency throughout the servlets and service!